### PR TITLE
[Mobile] Fix caret position after inline paste - after selection refactor

### PR DESCRIPTION
**Pull request by mkevins**
## Note
This PR is part of a series of PRs to fix the same issue. Most recently, a change in ownership of selection state has brought about an important refactor which regressed the previous PR before it was merged: https://github.com/WordPress/gutenberg/pull/15552. The solution there is no longer viable, but the unit test in that PR may still serve as a helpful tool to prevent more regressions.

## Description
This PR is intended to fix a regression of _part_ of this issue: https://github.com/wordpress-mobile/gutenberg-mobile/issues/828 , specifically, when inline content is pasted.

[Companion PR](https://github.com/wordpress-mobile/gutenberg-mobile/pull/995).

## How has this been tested?
This has been tested using the steps here:
https://github.com/wordpress-mobile/gutenberg-mobile/issues/828#issuecomment-481546078

## Types of changes
This is a bug fix, but currently only resolves a _part_ of the original issue. It serves as an incremental improvement. Also, this PR renames some local variables for clarity.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate.

### CODE REVIEW

File: packages/block-editor/src/components/rich-text/index.native.js [hunk 1]
```diff
@@ -489,14 +489,14 @@ export class RichText extends Component {
...
+			const resultingRecord = insert( currentRecord, recordToInsert );
+			const resultingContent = this.valueToFormat( resultingRecord );
 
 			this.lastEventCount = undefined;
-			this.value = newContent;
+			this.value = resultingContent;
 
 			// explicitly set selection after inline paste
-			this.forceSelectionUpdate( insertedContent.start, insertedContent.end );
+			this.onSelectionChange( resultingRecord.start, resultingRecord.end );
```
Comment (mkevins) [inline packages/block-editor/src/components/rich-text/index.native.js]: @Tug can you confirm whether this is the right approach to update the selection after the inline paste event?

Comment (Tug) [reply packages/block-editor/src/components/rich-text/index.native.js]: This looks good to me 👍 
It seems we don't need `forceSelectionUpdate` anymore, right?

Comment (mkevins) [reply packages/block-editor/src/components/rich-text/index.native.js]: Good catch! I pushed a commit to remove that function.

File: packages/block-editor/src/components/rich-text/index.native.js [hunk 2]
```diff
@@ -673,13 +673,6 @@ export class RichText extends Component {
 		return value;
 	}
 
-	componentWillReceiveProps( nextProps ) {
```
Comment (mkevins) [inline packages/block-editor/src/components/rich-text/index.native.js]: I've merged master and resolved a conflict. `componentWillReceiveProps` was removed after I diverged from master: https://github.com/WordPress/gutenberg/commit/27a6ead99b6ec38106c2b8696c74fe2a6e3d24d2#diff-4828a21853e899e5a36faecfa96d83e8L613.

**Review by Tug — APPROVED** [review]
LGTM 👍

**Pull request merged**